backend/apps/authentication/views.py

```
@@ -10,19 +10,154 @@
+        refresh_token = request.COOKIES.get("refresh_token")
+
+        if not refresh_token:
+            return Response(status=status.HTTP_400_BAD_REQUEST)
+
+        token = RefreshToken(refresh_token)
+        try:
+            token.blacklist()
+            response = Response(status=status.HTTP_204_NO_CONTENT)
+
+            response.delete_cookie("refresh_token")
+            response.delete_cookie("access_token")
+
+            return response
+        except Exception:
+            return Response(status=status.HTTP_400_BAD_REQUEST)
+
+
+class ResendVerificationView(APIView):
+    """
+    Generates access token for link,
+    sends mail with verification link.
+    """
+
+    throttle_classes = [CommonRedisThrottle]
+
+    def post(self, request):
+        email = request.data.get("email")
+
+        if not email:
+            return Response(
+                {"detail": "Necessary fields are missing."},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+        try:
+            user = User.objects.get(email=email)
+
+            if user.is_active:
+                return Response(
+                    {"detail": "Account was already verified"},
+                    status=status.HTTP_400_BAD_REQUEST,
+                )
+
+        except User.DoesNotExist:
+            return Response(status=status.HTTP_200_OK)  # 400 for test
+
+        token = default_token_generator.make_token(user)
+        uid = urlsafe_base64_encode(force_bytes(user.id))
+
+        verification_link = (
+            f"{settings.FRONTEND_URL}/api/auth/verify-email/{uid}/{token}/"
+        )
+
+        try:
+            send_mail(
+                subject="Verify your email",
+                message=f"Please, verify your email by clicking: {verification_link}",
+                from_email=settings.DEFAULT_FROM_EMAIL,
+                recipient_list=[user.email],
+                fail_silently=False,
+            )
+        except Exception:
```

except Exception: is used for mail sending; this swallows detailed failures and only logs a generic message. It may mask configuration issues (SMTP auth errors, invalid recipient, template errors).

Catch specific exceptions (e.g., smtplib.SMTPException) and return a 5xx response if sending fails in a critical flow, or at least log the exception message and include context.
```
except Exception as e:
    logger.exception("Failed to send verification email: %s", e)
    return Response({"detail": "Failed to send email"}, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
```

---

Done.